lib/pixel-mapper.cc

```diff
@@ -237,24 +245,39 @@ class VerticalMapper : public PixelMapper {
     *visible_width = matrix_width * parallel_ / chain_;
     *visible_height = matrix_height * chain_ / parallel_;
 #if 0
-    fprintf(stderr, "%s: C:%d P:%d. Turning W:%d H:%d Physical "
-            "into W:%d H:%d Virtual\n",
-            GetName(), chain_, parallel_,
-            *visible_width, *visible_height, matrix_width, matrix_height);
+     fprintf(stderr, "%s: C:%d P:%d. Turning W:%d H:%d Physical "
+	     "into W:%d H:%d Virtual\n",
+             GetName(), chain_, parallel_,
+	     *visible_width, *visible_height, matrix_width, matrix_height);
 #endif
     return true;
   }
 
   virtual void MapVisibleToMatrix(int matrix_width, int matrix_height,
                                   int x, int y,
                                   int *matrix_x, int *matrix_y) const {
-    int panel_width  = matrix_width  / chain_;
-    int panel_height = matrix_height / parallel_;
+    const int panel_width  = matrix_width  / chain_;
+    const int panel_height = matrix_height / parallel_;
+
     *matrix_x = (x % panel_width) +  int(y/panel_height)* panel_width;
     *matrix_y = (y % panel_height) + int(x/panel_width) * panel_height;
+
+    if (z_) {
```

Comment:
The cyclomatic complexity is still there because you have two nested ifs. But if you move the offset_cnt alculation out, you can have the if-condition in one expression

```c++
const int x_panel_offset_cnt = *matrix_x / panel_width;
if (z_ && x_panel_offset_cnt % 2 == 1) {
  ...
}
```

Reply:
Sure, I can do that. It does create one extra division in the case of panels with no inversion, but we agreed that efficiency is not the main problem, so if you prefer it as a single if, done.
And I did try to move that const up with the other ones above. I know some people object to having variable declarations in the middle of a block, but if it is to remain a const, it's unavoidable.

Reply:
Done